tests/security/test_repo_setup.pm

```diff
@@ -0,0 +1,107 @@
+# Copyright © 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Setup online repos or dvd images for further testing
+# Maintainer: wnereiz <wnereiz@member.fsf.org>
+# Tags: poo#52808
+
+use strict;
+use warnings;
+use base "consoletest";
+use testapi;
+use utils 'zypper_call';
+
+sub repo_cleanup {
+    assert_script_run "SUSEConnect --cleanup";
+    assert_script_run "rm -f /etc/zypp/repos.d/*.repo";
+    # Make sure all repositories are cleaned
+    validate_script_output("zypper lr || true", sub { m/No repo/ });
+}
+
+sub run {
+    my ($self) = @_;
+    select_console 'root-console';
+
+    # Note: Check the ticket in p.o.o for detailed descriptions of the
+    #       repository setup logic.
+
+    if (!get_var('SECTEST_DVD_SRC')) {
+        my $sc_out = script_output("SUSEConnect --status-text", proceed_on_failure => 1);
+
+        record_info "Not Registered", "System is not registered, add DVD image as repositories." if ($sc_out =~ /Not Registered/);
+
+        # If the the network is not available to check if it is registered.
+        # Then proceed following steps to add DVD repo.
+        record_info "Network Error", "SUSEConnect error: SocketError" if ($sc_out =~ /SocketError/);
+
+        if ($sc_out =~ /Status.*ACTIVE/) {
+            my $regcode = get_var("SCC_REGCODE");
+            ($regcode) = $sc_out =~ /Regcode: (.*)/ if (!$regcode);
+            die "SCC_REGCODE not specified" if (!$regcode);
+            repo_cleanup;
+            # Base system registration
+            assert_script_run("SUSEConnect -r $regcode", timeout => 180);
+
+            # WE addon registration (according to the conditions)
+            if (get_var("SECTEST_REQUIRE_WE") && check_var("ARCH", "x86_64")) {
+                my $regcode_we = get_required_var("SCC_REGCODE_WE");
+                my ($pd_no) = $sc_out =~ /\(SLES\/(.*)\/x86_64\)/;
+
+                # Register Desktop module as a dependency
+                assert_script_run("SUSEConnect -p sle-module-desktop-applications/$pd_no/x86_64", timeout => 180);
+
+                # Register WE with workaround if NVIDIA repository is (usually) not ready
+                die "WE registration failed" if
+                  (script_output("SUSEConnect -p sle-we/$pd_no/x86_64 -r $regcode_we",
```

Comment:
Same here, invoke function add_suseconnect_product instead assert_script_run.

Reply:
Need to check the stdout so we do not use `add_suseconnect_product` here.